### ISSUE TITLE
FreeSurfaceApplication linking problem on windows

### ISSUE THREAD

**Issue by roigcarlo** [KratosMultiphysics/Kratos#6210]
The application cannot compile due to a linking problem with a GidIO class:

```C++
namespace Kratos
{
namespace Python
{
typedef GidIO<EdgebasedGidGaussPointsContainer,EdgebasedGidMeshContainer> GidIOType; // This is causing problems
typedef GidIO<> GidIOBaseType;

void WriteNodeMesh( GidIOType& dummy, GidIOType::MeshType& rThisMesh )
{
```

An explicit instantiation of the class must be given somewhere in the class or its definition will not be available at link time.

Ping @joaquinirazabal, as I am not aware who is maintaining this and you appear in most commits.

**Comment by philbucher** [comment]
probably related to #5653 
@maceligueta

**Comment by maceligueta** [comment]
Thanks @philbucher, but in #5653 and #5868 the code was compiling and (apparently) linking well, no errors. It did not run as expected, though. 
In this case I would create an explicit instantiation of the new templated class that is being used, as @roigcarlo recommends...

**Comment by joaquinirazabal** [comment]
I have just created a PR #6212, but I think I need some help as I never faced this kind of problem. I also tag @antonialarese as she is also in charge of the application.

**Comment by roigcarlo** [comment]
This should have been solved in #6212, closing, reopen if needed
